### PR TITLE
Add example for Array.flatMap()

### PR DESCRIPTION
**Pull request by talitaoliveira**
Pull Request for issue #1587

### CODE REVIEW

File: live-examples/js-examples/array/array-flatmap.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const humans = [
+  {
+    name: 'Human 1',
+    phones: [
+      { type: 'cellphone', number: '9.9999-9999' },
+      { type: 'landline', number: '3444-4444' },
+    ],
+  },
+  {
+    nome: 'Human 2',
```
Comment (krankj) [inline live-examples/js-examples/array/array-flatmap.js:10]: typo "nome"

Comment (talitaoliveira) [reply live-examples/js-examples/array/array-flatmap.js]: Sorry! Fixed it.

**Review by velusgautam — CHANGES_REQUESTED** [review]
Just a suggestion
Can you give some more simple dataset example to show the flatmap process. It first map() and then it flat() to one level deep. 

const nums = [1,2, [3], [4],[] ];
const flattened = nums.flatMap(num => num);
// [ 1, 2, 3, 4 ]

**Comment by talitaoliveira** [comment]
Hey @velusgautam . Done with your suggestion.
I think that's valid, but when I made the first example the intention was to be more a "real life example".
But that's ok. :)

**Comment by chrisdavidmills** [comment]
This is looking pretty good, thanks @talitaoliveira ! It looks like a bunch of rogue commits have ended up in this PR though; I'm assuming you didn't touch 16 files. A rebase against latest origin would probably fix this?

**Comment by talitaoliveira** [comment]
Yeah I didn't touch them. Probably I did something wrong when getting the changes from the origin.
Now I guess it's right...

**Comment by tannerdolby** [comment]
@wbamberg Is this PR still under review or has it become stale? I have an Array.flatMap() interactive example spun up in my local `mdn/interactive-examples` repo if you want me to submit that as a fresh PR. Let me know!

**Comment by wbamberg** [comment]
> @wbamberg Is this PR still under review or has it become stale? I have an Array.flatMap() interactive example spun up in my local `mdn/interactive-examples` repo if you want me to submit that as a fresh PR. Let me know!

@tannerdolby , this PR still seems to have lots of extra changes in it, so yes, please do submit a new one. Thanks!

**Comment by Rumyra** [comment]
Closing this pr due to it being superseeded by https://github.com/mdn/interactive-examples/pull/1845

**Pull request closed without merging**